### DISCUSSION TITLE
Session not persisting

### DISCUSSION THREAD

**Discussion by dmatthams** [Intevel/nuxt-directus#174] [Q&A] [answered]
Hi! I've setup nuxt-directus and can successfully login + the cookies get set, yet when refreshing the page or closing the window and returning it seems to lose the session. I have `autoFetch: true` Am I missing something?

**Comment by Intevel** [comment]
Is the cookie set after closing the window?

**Reply by dmatthams** [reply]
Yes after closing tab the session + refresh cookies remain.

**Reply by Intevel** [reply]
Do you have a middleware?

**Reply by dmatthams** [reply]
Yes, the same as the example:

```
export default defineNuxtRouteMiddleware((to, _from) => {
    const user = useDirectusUser();
    if (!user.value) {
        return navigateTo("/");
    }
});
```

**Reply by dmatthams** [reply]
Ok, getting this error in console:

```
[Directus Error]: FetchError: request to http://localhost:8055/users/me failed, reason: connect ECONNREFUSED 127.0.0.1:8055 (http://localhost:8055/users/me)
Couldn't fetch user TypeError: Cannot read properties of undefined (reading '_data')
```

**Comment by Adashi12** [comment]
I also have the problem that the session seems not persisting if I put a middleware like described in the docs:

`export default defineNuxtRouteMiddleware((to, _from) => {
const user = useDirectusUser();  
if (!user.value) {    
return navigateTo("/login");  
}
});`

I get no errors in the console though.

**Reply by casualmatt** [reply, marked as answer]
this should work as middleware, I will update the documentation soon. 
```ts
export default defineNuxtRouteMiddleware(async (to, _from) => {
  const { fetchUser, setUser } = useDirectusAuth();
  const user = useDirectusUser();
  if (!user.value) {
    const user = await fetchUser();
    setUser(user.value);
  }
  if (!user.value) {
    return navigateTo("/login");
  }
});
```

**Reply by Nisthar** [reply]
@casualmatt 
![image](https://github.com/user-attachments/assets/f708235d-523c-4d47-a190-acbe5900f53f)
why is my user.value always null? if i login just now and hit the middleware, its still null.

**Reply by Nisthar** [reply]
both user from useDirectusUser() and fetchUser() is undefined.
